### ISSUE TITLE
Attendee information upload tool for CampTix

### ISSUE THREAD

**Issue by ghost** [WordPress/wordcamp.org#591]
Imported from https://meta.trac.wordpress.org/ticket/2974 
Created by @andreamiddleton:

Occasionally WordCamps must run their registration through non-WordCamp channels, for various reasons. 

This is a challenge for the WordCamp program, because we lose access to attendance and attendee data. This affects our ability to report accurately on WordCamp programs to sponsors and the global community. 

A simple attendee upload tool would resolve this challenge. It should allow organizers to upload a CSV file to CampTix with the following attendee information, at a minimum:

First Name
Last Name
Email Address
Ticket price
Date of purchase

Organizers could create a ticket with question fields that match the CSV file they have from the other registration platform. Then, they could navigate to the Tools section and choose the Import Attendee Data tab.

On that tab, they'd find a selector for the ticket they wanted to map the attendee data to, then a file selector. Upon selecting the file, the tool would provide suggested mapping based on CSV file headings, if there were any, and otherwise ask the user to map the columns to the ticket fields. 

Once the mapping was confirmed, the data would be viewable in Tickets > Attendees

**Comment by ghost** [comment]
Comment by @andreamiddleton:
 * [Screen Shot 2017-07-18 at 9.21.23 PM.png](https://meta.trac.wordpress.org/attachment/ticket/2974/Screen%20Shot%202017-07-18%20at%209.21.23%20PM.png) added

A very basic wireframe for the proposed Import Attendee Data tab on Tickets > Tools

**Comment by ghost** [comment]
[Comment](https://meta.trac.wordpress.org/ticket/2974#comment:1) by @hlashbrooke:

I maybe wrong (as CampTix and WordCamp.org are blurred a bit), but I think the attendee side of things is all handled inside CampTix itself, so this would need to be discussed and worked in the CampTix Githhub repo: https://github.com/Automattic/camptix. I could be horrendously mistaken here though, so ignore me if that's the case :P

**Comment by ghost** [comment]
[Comment](https://meta.trac.wordpress.org/ticket/2974#comment:2) by @coreymckrill:
 * Keywords set to `Has Screenshots` `Needs Patch` `Good First Bug`
 * Status set to `accepted`
 * Owner set to @coreymckrill

This could be an enhancement for CampTix itself, or it could be built as a separate plugin that works with CampTix (like the badge generator and the network tools). So you're both right. My inclination would be to build it as a separate plugin, if CampTix has the necessary integration points. So, I think it's ok to leave this here in Trac.

**Comment by ghost** [comment]
[Comment](https://meta.trac.wordpress.org/ticket/2974#comment:3) by @coreymckrill:
 * Status set to `assigned`
 * Owner cleared

**Comment by ghost** [comment]
[Comment](https://meta.trac.wordpress.org/ticket/2974#comment:4) by [@khromov](https://profiles.wordpress.org/khromov/):

@coreymckrill Hi, I wanted to see if there has been any movement on this ticket. We had our WordCamp recently using an external ticketing system and we promised @andreamiddleton to transfer the names over into CampTix, but we're not sure we will have the manpower to pull the proposed PR off by ourselves. A simpler option for us would be if the REST API was available, we could write a simple script to upload the participants via the API.

**Comment by ghost** [comment]
[Comment](https://meta.trac.wordpress.org/ticket/2974#comment:5) by @iandunn:

I think a CSV upload is going to be a better choice overall, because it'll be more accessible to most camps. Most camps won't want to write a custom script, even if they have a dev with time to do it.

**Comment by ghost** [comment]
[Comment](https://meta.trac.wordpress.org/ticket/2974#comment:6) by slackbot:

_This ticket was mentioned in [Slack](https://make.wordpress.org/chat/) in #meta-wordcamp by ryelle. [View the logs](https://wordpress.slack.com/archives/meta-wordcamp/p1584639974112800)._

**Comment by timiwahalahti** [comment]
I don't remember any recent WordCamps selling tickets elsewhere than our systems. @angelasjin and @dorsvenabili do you remember cases of external ticketing after 2017?

If not, I'd say it's safe to close this issue.

**Comment by dorsvenabili** [comment]
Many WordCamps don't use our ticket system, some examples: WC Kampala, WC Brno,... and more. They don't use Camptix because the normal method of payments in their countries are not covered by our plugins.

So, having a system to upload the attendees info would be very useful and welcome for our stats.

Thanks!

**Comment by harmonyromo** [comment]
I agree with Rocio here. Brno is definitely one that has never run ticket sales through Camptix. I also know that some of the WordCamps that have happened in 2023, Iloilo for example, run some ticket sales through Camptix and some locally, so the ticket totals shown in our reporting only capture some of the sales. Granted, there are not a large number of events that would need this option, but if it's not a huge lift, having an import option would be helpful to improve our data accuracy.
